haystack/nodes/prompt/prompt_node.py

```diff
@@ -0,0 +1,822 @@
+    ),
+    PromptTemplate(
+        name="question-answering-check",
+        prompt_text="Does the following context contain the answer to the question. "
+        "Context: $documents; Question: $questions; Please answer yes or no! Answer:",
+        prompt_params=["documents", "questions"],
+    ),
+    PromptTemplate(
+        name="sentiment-analysis",
+        prompt_text="Please give a sentiment for this context. Answer with positive, "
+        "negative or neutral. Context: $documents; Answer:",
+        prompt_params=["documents"],
+    ),
+    PromptTemplate(
+        name="multiple-choice-question-answering",
+        prompt_text="Question:$questions ; Choose the most suitable option to answer the above question. "
+        "Options: $options; Answer:",
+        prompt_params=["questions", "options"],
+    ),
+    PromptTemplate(
+        name="topic-classification",
+        prompt_text="Categories: $options; What category best describes: $documents; Answer:",
+        prompt_params=["documents", "options"],
+    ),
+    PromptTemplate(
+        name="language-detection",
+        prompt_text="Detect the language in the following context and answer with the "
+        "name of the language. Context: $documents; Answer:",
+    ),
+    PromptTemplate(
+        name="translation",
+        prompt_text="Translate the following context to $target_language. Context: $documents; Translation:",
+    ),
+]
+
+
+class PromptModelInvocationLayer:
+    """
+    PromptModelInvocationLayer implementations execute a prompt on an underlying model.
+
+    The implementation can be a simple invocation on the underlying model running in a local runtime, or
+    could be even remote, for example, a call to a remote API endpoint.
+    """
+
+    def __init__(self, model_name_or_path: str, max_length: Optional[int] = 100, **kwargs):
+        if model_name_or_path is None or len(model_name_or_path) == 0:
+            raise ValueError("model_name_or_path cannot be None or empty string")
+
+        self.model_name_or_path = model_name_or_path
+        self.max_length: Optional[int] = max_length
+
+    @abstractmethod
+    def invoke(self, *args, **kwargs):
+        pass
+
+    @classmethod
+    def supports(cls, model_name_or_path: str) -> bool:
+        return False
+
+
+class HFLocalInvocationLayer(PromptModelInvocationLayer):
+    """
+    A subclass of the PromptModelInvocationLayer class. It loads a pre-trained model from Hugging Face and
+    passes a prepared prompt into that model.
+
+    Note: kwargs other than init parameter names are ignored to enable reflective construction of the class
+    as many variants of PromptModelInvocationLayer are possible and they may have different parameters
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: str = "google/flan-t5-base",
+        max_length: Optional[int] = 100,
+        use_auth_token: Optional[Union[str, bool]] = None,
+        use_gpu: Optional[bool] = True,
```

Comment:
I would remove the `Optional` type here since `initialize_device_settings` requires `use_gpu` to be a bool. And giving the `Optional` type suggests that the value `None` would be a valid value for this variable.